### PR TITLE
[Vertex GA] Regenerate docs after Vertex GA changes

### PR DESCRIPTION
**Pull request by hsubox76**
Regenerated docs based on all code changes for Vertex GA so far.

Changes are:
- added Schema classes
- remove "unspecified" enum values
- change optionality on some properties

### CODE REVIEW

File: docs-devsite/vertexai-preview.generationconfig.md [hunk 1]
```diff
@@ -26,7 +26,8 @@ export interface GenerationConfig
 |  [frequencyPenalty](./vertexai-preview.generationconfig.md#generationconfigfrequencypenalty) | number |  |
 |  [maxOutputTokens](./vertexai-preview.generationconfig.md#generationconfigmaxoutputtokens) | number |  |
 |  [presencePenalty](./vertexai-preview.generationconfig.md#generationconfigpresencepenalty) | number |  |
-|  [responseMimeType](./vertexai-preview.generationconfig.md#generationconfigresponsemimetype) | string | Output response mimetype of the generated candidate text. Supported mimetypes are <code>text/plain</code> (default, text output) and <code>application/json</code> (JSON response in the candidates). The model needs to be prompted to output the appropriate response type, otherwise the behavior is undefined. This is a preview feature. |
+|  [responseMimeType](./vertexai-preview.generationconfig.md#generationconfigresponsemimetype) | string | Output response mimetype of the generated candidate text. Supported mimetypes are <code>text/plain</code> (default, text output) and <code>application/json</code> (JSON response in the candidates). |
```
Comment (rachelsaunders) [inline docs-devsite/vertexai-preview.generationconfig.md:29]: NOT BLOCKING

mimetype -> MIME type

(comment applies to several other instances, even in other files)

File: docs-devsite/vertexai-preview.generationconfig.md [hunk 2]
```diff
@@ -26,7 +26,8 @@ export interface GenerationConfig
...
+|  [responseSchema](./vertexai-preview.generationconfig.md#generationconfigresponseschema) | [TypedSchema](./vertexai-preview.md#typedschema) \| [SchemaRequest](./vertexai-preview.schemarequest.md#schemarequest_interface) | Output response schema of the generated candidate text. This value can be a class generated with a [Schema](./vertexai-preview.schema.md#schema_class) static method like <code>Schema.string()</code> or <code>Schema.object()</code> or it can be a plain JS object matching the [SchemaRequest](./vertexai-preview.schemarequest.md#schemarequest_interface) interface. Note: This only applies when the specified <code>responseMIMEType</code> supports a schema; currently this is limited to <code>application/json</code>. |
```
Comment (rachelsaunders) [inline docs-devsite/vertexai-preview.generationconfig.md:30]: NOT BLOCKING

Can the "Note: ..." be on its own line?

(comment applies to several other instances, even in other files)

File: docs-devsite/vertexai-preview.md [hunk 3]
```diff
@@ -25,8 +25,15 @@ The Vertex AI in Firebase Web SDK.
...
+|  [BooleanSchema](./vertexai-preview.booleanschema.md#booleanschema_class) | Schema class for "boolean" types. |
 |  [ChatSession](./vertexai-preview.chatsession.md#chatsession_class) | ChatSession class that enables sending chat messages and stores history of sent and received messages so far. |
 |  [GenerativeModel](./vertexai-preview.generativemodel.md#generativemodel_class) | Class for generative model APIs. |
+|  [IntegerSchema](./vertexai-preview.integerschema.md#integerschema_class) | Schema class for "integer" types. |
+|  [NumberSchema](./vertexai-preview.numberschema.md#numberschema_class) | Schema class for "number" types. |
+|  [ObjectSchema](./vertexai-preview.objectschema.md#objectschema_class) | Schema class for "object" types. The <code>properties</code> param must be a map of Schema. |
+|  [Schema](./vertexai-preview.schema.md#schema_class) | Parent class encompassing all Schema types, with static methods that allow building specific Schema types. This class can be converted with JSON.stringify() into a JSON string accepted by Vertex REST endpoints. (This string conversion is automatically done when calling SDK methods.) |
```
Comment (rachelsaunders) [inline docs-devsite/vertexai-preview.md:35]: NOT BLOCKING

JSON.stringify() should be tagged as code

(comment applies to several other instances, even in other files)

Comment (rachelsaunders) [inline docs-devsite/vertexai-preview.md:35]: Vertex -> Vertex AI

(comment applies to several other instances, even in other files)

File: docs-devsite/vertexai-preview.md [hunk 4]
```diff
@@ -36,12 +43,12 @@ The Vertex AI in Firebase Web SDK.
 |  [BlockReason](./vertexai-preview.md#blockreason) | Reason that a prompt was blocked. |
 |  [FinishReason](./vertexai-preview.md#finishreason) | Reason that a candidate finished. |
 |  [FunctionCallingMode](./vertexai-preview.md#functioncallingmode) |  |
-|  [FunctionDeclarationSchemaType](./vertexai-preview.md#functiondeclarationschematype) | Contains the list of OpenAPI data types as defined by https://swagger.io/docs/specification/data-models/data-types/ |
 |  [HarmBlockMethod](./vertexai-preview.md#harmblockmethod) |  |
 |  [HarmBlockThreshold](./vertexai-preview.md#harmblockthreshold) | Threshold above which a prompt or candidate will be blocked. |
 |  [HarmCategory](./vertexai-preview.md#harmcategory) | Harm categories that would cause prompts or candidates to be blocked. |
 |  [HarmProbability](./vertexai-preview.md#harmprobability) | Probability that a prompt or candidate matches a harm category. |
 |  [HarmSeverity](./vertexai-preview.md#harmseverity) | Harm severity levels. |
+|  [SchemaType](./vertexai-preview.md#schematype) | Contains the list of OpenAPI data types as defined by https://swagger.io/docs/specification/data-models/data-types/ |
```
Comment (rachelsaunders) [inline docs-devsite/vertexai-preview.md:51]: NOT BLOCKING

Usually, we try to make all URLs into hyperlinks rather than having the actual URL exposed.

(comment applies to several other instances, even in other files)

File: docs-devsite/vertexai-preview.md [hunk 5]
```diff
@@ -80,11 +85,16 @@ The Vertex AI in Firebase Web SDK.
...
 |  [RetrievedContextAttribution](./vertexai-preview.retrievedcontextattribution.md#retrievedcontextattribution_interface) |  |
 |  [SafetyRating](./vertexai-preview.safetyrating.md#safetyrating_interface) | A safety rating associated with a [GenerateContentCandidate](./vertexai-preview.generatecontentcandidate.md#generatecontentcandidate_interface) |
 |  [SafetySetting](./vertexai-preview.safetysetting.md#safetysetting_interface) | Safety setting that can be sent as part of request parameters. |
+|  [SchemaInterface](./vertexai-preview.schemainterface.md#schemainterface_interface) | Interface for Schema class. |
+|  [SchemaParams](./vertexai-preview.schemaparams.md#schemaparams_interface) | User-facing params passed to specific Schema static methods. |
+|  [SchemaRequest](./vertexai-preview.schemarequest.md#schemarequest_interface) | Final format for Schema params passed to backend requests. |
+|  [SchemaShared](./vertexai-preview.schemashared.md#schemashared_interface) | Basic <code>Schema</code> properties shared across several Schema-related types. |
```
Comment (rachelsaunders) [inline docs-devsite/vertexai-preview.md:97]: Why is Schema tagged as code here, but not anywhere else?

FWIW - I would prolly recommend tagging it almost everywhere as code.  And ObjectSchema and StringSchema, too.

File: docs-devsite/vertexai-preview.schema.md [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+|  --- | --- | --- |
+|  [(constructor)(schemaParams)](./vertexai-preview.schema.md#schemaconstructor) |  | Constructs a new instance of the <code>Schema</code> class |
+
+## Properties
+
+|  Property | Modifiers | Type | Description |
+|  --- | --- | --- | --- |
+|  [description](./vertexai-preview.schema.md#schemadescription) |  | string | Optional. The description of the property. |
+|  [example](./vertexai-preview.schema.md#schemaexample) |  | unknown | Optional. The example of the property. |
+|  [format](./vertexai-preview.schema.md#schemaformat) |  | string | Optional. The format of the property. Supported formats: for NUMBER type: "float", "double" for INTEGER type: "int32", "int64" for STRING type: "email", "byte", etc |
```
Comment (rachelsaunders) [inline docs-devsite/vertexai-preview.schema.md:34]: These need to be separated rather than running on to each other (or be broken out into a bullet list if that's possible).

for NUMBER type: "float", "double"
for INTEGER type: "int32", "int64"
etc.

File: docs-devsite/vertexai-preview.schema.md [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+|  Method | Modifiers | Description |
+|  --- | --- | --- |
+|  [array(arrayParams)](./vertexai-preview.schema.md#schemaarray) | <code>static</code> |  |
+|  [boolean(booleanParams)](./vertexai-preview.schema.md#schemaboolean) | <code>static</code> |  |
+|  [enumString(stringParams)](./vertexai-preview.schema.md#schemaenumstring) | <code>static</code> |  |
+|  [integer(integerParams)](./vertexai-preview.schema.md#schemainteger) | <code>static</code> |  |
+|  [number(numberParams)](./vertexai-preview.schema.md#schemanumber) | <code>static</code> |  |
+|  [object(objectParams)](./vertexai-preview.schema.md#schemaobject) | <code>static</code> |  |
+|  [string(stringParams)](./vertexai-preview.schema.md#schemastring) | <code>static</code> |  |
+|  [toJSON()](./vertexai-preview.schema.md#schematojson) |  | Defines how this Schema should be serialized as JSON. See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global\_Objects/JSON/stringify\#tojson\_behavior |
```
Comment (rachelsaunders) [inline docs-devsite/vertexai-preview.schema.md:49]: another exposed URL that should be a hyperlink

File: docs-devsite/vertexai-preview.schemaparams.md [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+
+{% comment %}
+DO NOT EDIT THIS FILE!
+This is generated by the JS SDK team, and any local changes will be
+overwritten. Changes should be made in the source code at
+https://github.com/firebase/firebase-js-sdk
+{% endcomment %}
+
+# SchemaParams interface
+User-facing params passed to specific Schema static methods.
```
Comment (rachelsaunders) [inline docs-devsite/vertexai-preview.schemaparams.md:13]: By "user", do you mean end-user?

**Review by rachelsaunders — CHANGES_REQUESTED** [review]

File: docs-devsite/vertexai-preview.schemaparams.md (hunk 8 above)
Comment (hsubox76) [reply docs-devsite/vertexai-preview.schemaparams.md:13]: Developer, as in, not internal and only used by other internal code. I can reword this to not be about who it's facing, but about if you're using the `Schema` convenience classes or looking at the raw request format.

File: docs-devsite/vertexai-preview.generationconfig.md (hunk 2 above)
Comment (hsubox76) [reply docs-devsite/vertexai-preview.generationconfig.md:30]: It's a little weird, I have to put a `<br />` into the code comment, but it works.

File: docs-devsite/vertexai-preview.generationconfig.md (hunk 1 above)
Comment (hsubox76) [reply docs-devsite/vertexai-preview.generationconfig.md]: Fixed.

File: docs-devsite/vertexai-preview.md (hunk 3 above)
Comment (hsubox76) [reply docs-devsite/vertexai-preview.md]: Fixed.

Comment (hsubox76) [reply docs-devsite/vertexai-preview.md]: Fixed.

File: docs-devsite/vertexai-preview.md (hunk 4 above)
Comment (hsubox76) [reply docs-devsite/vertexai-preview.md]: Fixed.

File: docs-devsite/vertexai-preview.md (hunk 5 above)
Comment (hsubox76) [reply docs-devsite/vertexai-preview.md]: Can actually make them links, so did so.

File: docs-devsite/vertexai-preview.schema.md (hunk 6 above)
Comment (hsubox76) [reply docs-devsite/vertexai-preview.schema.md]: Inserted html list into the comment.

File: docs-devsite/vertexai-preview.schema.md (hunk 7 above)
Comment (hsubox76) [reply docs-devsite/vertexai-preview.schema.md]: Fixed.

**Review by hsubox76 — COMMENTED** [review]
Addressed comments. Some of the changes have added some source files to this PR, that are the source of the changes. You don't have to review them, you can just look at the markdown files.

**Review by DellaBitta — COMMENTED** [review]
Is it a problem that `docs-devsite/_toc.yaml` was removed?

File: docs-devsite/vertexai.generationconfig.md [hunk 9]
```diff
@@ -66,14 +66,24 @@ presencePenalty?: number;
...
 
 <b>Signature:</b>
 
 ```typescript
 responseMimeType?: string;
 ```
 
+## GenerationConfig.responseSchema
+
+Output response schema of the generated candidate text. This value can be a class generated with a [Schema](./vertexai-preview.schema.md#schema_class) static method like `Schema.string()` or `Schema.object()` or it can be a plain JS object matching the [SchemaRequest](./vertexai-preview.schemarequest.md#schemarequest_interface) interface. <br/>Note: This only applies when the specified `responseMIMEType` supports a schema; currently this is limited to `application/json`<!-- -->.
```
Comment (rachelsaunders) [inline docs-devsite/vertexai.generationconfig.md:79]: Shouldn't `responseSchema` also support the MIME type of `text/x.enum`?

https://cloud.google.com/vertex-ai/generative-ai/docs/multimodal/control-generated-output#considerations

Comment (rachelsaunders) [inline docs-devsite/vertexai.generationconfig.md:79]: NOT BLOCKING

Schema and SchemaRequest

Both should be tagged as code in the source code. Pls look throughout the source code for other instances.


Anytime that you're referring to a specific component (method / field / enum / etc. name, file names, input / output value, etc.), then tag it as code.

One very generalized way to think about it to decide if something should be tagged as code: "Would I want this translated into German, Korean, etc.?" Note that our ref docs aren't translated, but this way of thinking helps you decide about appropriate tagging :-)

File: docs-devsite/vertexai-preview.md (hunk 4 above)
Comment (rachelsaunders) [reply docs-devsite/vertexai-preview.md]: I can't seem to find the corresponding new file in the new set of files?

File: docs-devsite/vertexai-preview.md (hunk 5 above)
Comment (rachelsaunders) [reply docs-devsite/vertexai-preview.md]: I can't seem to find the corresponding new file in the new set of files?

**Review by rachelsaunders — CHANGES_REQUESTED** [review]

File: docs-devsite/vertexai.schemashared.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+
+|  Property | Type | Description |
+|  --- | --- | --- |
+|  [description](./vertexai.schemashared.md#schemashareddescription) | string | Optional. The description of the property. |
+|  [enum](./vertexai.schemashared.md#schemasharedenum) | string\[\] | Optional. The enum of the property. |
+|  [example](./vertexai.schemashared.md#schemasharedexample) | unknown | Optional. The example of the property. |
+|  [format](./vertexai.schemashared.md#schemasharedformat) | string | Optional. The format of the property. |
+|  [items](./vertexai.schemashared.md#schemashareditems) | T | Optional. The items of the property. |
+|  [nullable](./vertexai.schemashared.md#schemasharednullable) | boolean | Optional. Whether the property is nullable. |
+|  [properties](./vertexai.schemashared.md#schemasharedproperties) | { \[k: string\]: T; } | Optional. Map of Schemas. |
```
Comment (rachelsaunders) [inline docs-devsite/vertexai.schemashared.md:31]: NOT BLOCKING

This should _probably_ be this?

Map of `Schema` objects.

File: docs-devsite/vertexai-preview.md (hunk 3 above)
Comment (rachelsaunders) [reply docs-devsite/vertexai-preview.md]: I can't seem to find the corresponding new file in the new set of files?

File: docs-devsite/vertexai-preview.schema.md (hunk 7 above)
Comment (rachelsaunders) [reply docs-devsite/vertexai-preview.schema.md]: Was it intentional to remove `toJSON()` here?  Maybe I'm missing it, but it doesn't seem to be in the newly generated file?

docs-devsite/vertexai.schema.md

**Review by rachelsaunders — CHANGES_REQUESTED** [review]

File: docs-devsite/vertexai.generationconfig.md (hunk 9 above)
Comment (hsubox76) [reply docs-devsite/vertexai.generationconfig.md]: I'm assuming by "tag as code" you mean to make it so it ends up in the markdown within `<code>` tags? This was pretty tricky, we normally do it with backticks but you can't do it on links so you normally choose between either having a link or having code tags. I think I found a way to do it although it's a little fragile. I did a search/replace on all `@link`s in the package.

Comment (hsubox76) [reply docs-devsite/vertexai.generationconfig.md]: I guess so. Added. I don't really know anything about it other than what's in that page so I just put the type in there and didn't add any explanation. I originally pulled this comment off the proto. https://source.corp.google.com/piper///depot/google3/google/cloud/aiplatform/master/content.proto;l=269?q=generationconfig&ss=piper%2FGoogle%2FPiper:google3%2Fgoogle%2Fcloud%2Faiplatform%2Fmaster%2F 

I'm not sure what I'm supposed to use for the source of truth.

File: docs-devsite/vertexai-preview.schema.md (hunk 7 above)
Comment (hsubox76) [reply docs-devsite/vertexai-preview.schema.md]: Yes, it doesn't need to be publicly exposed, it's for internal use.

File: docs-devsite/vertexai.schemashared.md (hunk 10 above)
Comment (hsubox76) [reply docs-devsite/vertexai.schemashared.md:31]: Fixed.

File: docs-devsite/vertexai-preview.md (hunk 3 above)
Comment (hsubox76) [reply docs-devsite/vertexai-preview.md]: I think the missing files should be back now.

File: docs-devsite/vertexai.schemashared.md (hunk 10 above)
Comment (rachelsaunders) [reply docs-devsite/vertexai.schemashared.md:31]: I'm still seeing:

Map of Schemas.

instead of:

Map of `Schema` objects.

File: docs-devsite/vertexai.generationconfig.md (hunk 9 above)
Comment (rachelsaunders) [reply docs-devsite/vertexai.generationconfig.md]: Maybe I'm missing it, but I'm not seeing `text/x.enum` listed in this .md file as a possible MIME type?

**Review by rachelsaunders — CHANGES_REQUESTED** [review]

File: docs-devsite/vertexai.generationconfig.md (hunk 9 above)
Comment (hsubox76) [reply docs-devsite/vertexai.generationconfig.md]: I added it to `responseMimeType` but not `responseSchema`. It should be here now also.

File: docs-devsite/vertexai.generationconfig.md [hunk 11]
```diff
@@ -66,14 +67,24 @@ presencePenalty?: number;
...
 
 <b>Signature:</b>
 
 ```typescript
 responseMimeType?: string;
 ```
 
+## GenerationConfig.responseSchema
+
+Output response schema of the generated candidate text. This value can be a class generated with a `{@link Schema}` static method like `Schema.string()` or `Schema.object()` or it can be a plain JS object matching the <code>[SchemaRequest](./vertexai.schemarequest.md#schemarequest_interface)</code> interface. <br/>Note: This only applies when the specified `responseMIMEType` supports a schema; currently this is limited to `application/json`<!-- -->.
```
[the remark this answers is not in the document]
Comment (hsubox76) [reply docs-devsite/vertexai.generationconfig.md]: I re-ran the generator, I think all these are gone.

File: docs-devsite/vertexai.schemashared.md (hunk 10 above)
Comment (hsubox76) [reply docs-devsite/vertexai.schemashared.md]: I think I missed a second instance of it. Should be fixed now.

**Review by rachelsaunders — APPROVED** [review]
